src/construct.ts

```diff
@@ -417,21 +422,13 @@ export class Node {
       throw new Error(`Cannot add children to "${this.path}" during synthesis`);
     }
 
-    if (childName in this._children) {
+    if (this._children[childName]) {
       const name = this.id ?? '';
       const typeName = this.host.constructor.name;
       throw new Error(`There is already a Construct with name '${childName}' in ${typeName}${name.length > 0 ? ' [' + name + ']' : ''}`);
     }
 
-    if (!childName && this.id) {
-      throw new Error(`cannot add a nameless construct to the named scope: ${this.path}`);
-    }
-
     this._children[childName] = child;
-
-    if (Object.keys(this._children).length > 1 && Object.keys(this._children).filter(x => !x).length > 0) {
-      throw new Error('only a single construct is allowed in a scope if it has an empty name');
-    }
```

Comment:
I also don't think this branch can ever get called. Here was my reasoning:
1. From my other comment, `childName` will never take on a falsey value (like an empty string). 
2. From (1), the line `this._children[childName] = childl` can never add falsey values as keys to `this._children`.
3. There are no other lines of code that add new keys to `this._children`.
4. Therefore, `this._children` will always have keys with truthy values, so `Object.keys(this._children).filter(x => !x)` can never have a length greater than 0.